### PR TITLE
[YQL-17103] Support StartsWith with pg types in extract_predicate library

### PR DESCRIPTION
**Pull request by nepal**
- support StartsWith predicates for pg types

### CODE REVIEW

File: ydb/library/yql/core/common_opt/yql_co_simple1.cpp [hunk 1]
```diff
@@ -4665,6 +4665,16 @@ void RegisterCoSimpleCallables1(TCallableOptimizerMap& map) {
     map["IsDistinctFrom"] = std::bind(&OptimizeDistinctFrom<false>, _1, _2);
 
     map["StartsWith"] = map["EndsWith"] = map["StringContains"] = [](const TExprNode::TPtr& node, TExprContext& ctx, TOptimizeContext& /*optCtx*/) {
+        if (node->Head().GetTypeAnn()->GetKind() == ETypeAnnotationKind::Pg || node->Tail().GetTypeAnn()->GetKind() == ETypeAnnotationKind::Pg) {
```
Comment (vitstn) [inline ydb/library/yql/core/common_opt/yql_co_simple1.cpp:4668]: why this overload is needed?

**Review by vitstn — CHANGES_REQUESTED** [review]
need to discuss

File: ydb/library/yql/core/common_opt/yql_co_simple1.cpp (hunk 1 above)
Comment (nepal) [reply ydb/library/yql/core/common_opt/yql_co_simple1.cpp:4668]: This overload helps to simplify logic of predicate analysis. This way we make StartsWith with pg types to look exactly the same as with plain YQL types (no intermediate FromPg is needed), so the code which looks for StartsWith over Member() call will work without pg-specific tweaks.

Also, accepting PG types here looks like a nice feature to me.

**Comment by nepal** [comment]
> need to discuss

Discussed and decided
1) add test for StartsWith/EndsWith/StringContains with pg types
2) document new overload in docs

File: ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/basic/starts_ends_with.md [hunk 2]
```diff
@@ -4,23 +4,18 @@
...
+EndsWith(T str, U suffix)->Bool[?]
 ```
 
 Обязательные аргументы:
 
 * Исходная строка;
 * Искомая подстрока.
 
-Аргументы могут быть типов `String` или `Utf8` и могут быть опциональными.
+Аргументы должны иметь тип `String`/`Utf8` (или опциональный String`/`Utf8`) либо строковый PostgreSQL тип (`PgText`/`PgBinary`/`PgVarchar`).
```
Comment (vitstn) [inline ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/basic/starts_ends_with.md:17]: Pgbytea

**Review by vitstn — CHANGES_REQUESTED** [review]

File: ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/basic/starts_ends_with.md (hunk 2 above)
Comment (nepal) [reply ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/basic/starts_ends_with.md]: fixed

**Review by vitstn — APPROVED** [review]

**Pull request merged**